packages/amplify_auth_cognito/test/amplify_auth_cognito_confirmSignIn_test.dart

```diff
@@ -0,0 +1,107 @@
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:amplify_auth_cognito/amplify_auth_cognito.dart';
+import 'package:amplify_auth_cognito/method_channel_auth_cognito.dart';
+import 'package:amplify_core/amplify_core.dart';
+
+void main() {
+  const MethodChannel authChannel = MethodChannel('com.amazonaws.amplify/auth_cognito');
+  const MethodChannel coreChannel = MethodChannel('com.amazonaws.amplify/core');
+
+  Amplify amplify = new Amplify();
+  AmplifyAuthCognito auth = AmplifyAuthCognito();
+
+  TestWidgetsFlutterBinding.ensureInitialized();
+
+  setUp(() {
+    coreChannel.setMockMethodCallHandler((MethodCall methodCall) async {
+      return true;
+    });
+  });
+
+  tearDown(() {
+    authChannel.setMockMethodCallHandler(null);
+    coreChannel.setMockMethodCallHandler(null);
+  });
+
+  test('confirmSignIn request returns a SignInResult', () async {
+
+ authChannel.setMockMethodCallHandler((MethodCall methodCall) async {
+    if (methodCall.method == "confirmSignIn") {
+      assert(methodCall.arguments["data"] is Map);
+      assert(methodCall.arguments["data"]["confirmationCode"] is String);
+          return {
+            "isSignedIn": false,
+            "nextStep": {
+              "signInStep": "DONE",
+              "codeDeliveryDetails":  {
+                "deliveryMedium": "EMAIL",
+                "attributeName": "email",
+                "destination": "test@test.test"
+              }
+            }
+          };
+    } else {
+      return true;
+    }     
+  });
+
+    await amplify.addPlugin(authPlugins: [auth]);
+    await amplify.configure("{}");
+    var expectation = CognitoSignInResult(
+      isSignedIn: false, 
+      nextStep: AuthNextSignInStep(
+        additionalInfo: {},
+        codeDeliveryDetails: {
+          "deliveryMedium": "EMAIL",
+          "attributeName": "email",
+          "destination": "test@test.test"
+        },
+        signInStep: "DONE",
+      )
+    );
+    var res = await Amplify.Auth.confirmSignIn(confirmationValue: "iAmLegit");
+    expect(res.isSignedIn, equals(expectation.isSignedIn));
+    expect(res.nextStep.signInStep, equals(expectation.nextStep.signInStep));
+    expect(res.nextStep.additionalInfo, equals(expectation.nextStep.additionalInfo));
+    expect(res.nextStep.codeDeliveryDetails.attributeName, equals(expectation.nextStep.codeDeliveryDetails.attributeName));
+    expect(res.nextStep.codeDeliveryDetails.deliveryMedium, equals(expectation.nextStep.codeDeliveryDetails.deliveryMedium));
+    expect(res.nextStep.codeDeliveryDetails.destination, equals(expectation.nextStep.codeDeliveryDetails.destination));
```

Comment:
instead of having many smaller `expect` checks, can you create an expected `res`, and then do a single `expect(expectedRes, equals(actualRes));`?

Reply:
I don't this think this works in Dart due to object hashCodes.

Reply:
Gotcha, I think this is the same situation as occurs in Java, then, where you have to implement your own equals and hashCode? https://stackoverflow.com/a/52602964/695787

Reply:
You could use extensions (even in the test classes) to add a `toString()` or `equals()` method if you don't want to pollute data classes.

Reply:
@Amplifiyer @jamesonwilliams - I tried my hand at an extension,